integ/src/eks_provider.rs

```diff
@@ -124,9 +132,71 @@ pub async fn get_cluster_info(cluster_name: &str, region: &str) -> ProviderResul
         controlplane_sg,
         clustershared_sg,
         iam_instance_profile_arn,
+        dns_ip,
     })
 }
 
+async fn dns_ip(eks_client: &aws_sdk_eks::Client, cluster_name: &str) -> ProviderResult<String> {
+    let describe_results = eks_client
+        .describe_cluster()
+        .name(cluster_name)
+        .send()
+        .await
+        .context("Unable to get eks describe cluster")?;
+
+    let kubernetes_network_config = describe_results
+        .cluster
+        .as_ref()
+        .context("Response missing cluster field")?
+        .kubernetes_network_config
+        .as_ref()
+        .context("Cluster missing kubernetes_network_config field")?;
+
+    let ip_family = kubernetes_network_config
+        .ip_family
+        .as_ref()
+        .context("IP family missing data")
+        .map(|ids| ids.clone())?;
+
+    match ip_family {
+        IpFamily::Ipv4 => {
+            let ipv4_cidr = kubernetes_network_config
+                .service_ipv4_cidr
+                .as_ref()
+                .context("Ipv6 idr missing data")
+                .map(|ids| ids.clone())?;
+
+            let ipv4 = transform_dns_ip(&ipv4_cidr, IPV4_DIVIDER, IPV4_OCTET);
```

Comment:
Cluster DNS IP conversion for IPv4 clusters is slightly more complicated. See https://github.com/awslabs/amazon-eks-ami/blob/02823975992a9291609498b00c00b6434e0e4d19/files/bootstrap.sh#L387-L408

If the service ipv4 cidr, starts with a `10` for the first octet, then we return `10.100.0.10`. Otherwise, we return `172.20.0.10`. We have the same logic in pluto: 
https://github.com/bottlerocket-os/bottlerocket/blob/62c2e64df8885c88534624288630c69342a589be/sources/api/pluto/src/main.rs#L214-L216

Reply:
Sorry, correction. If service IPv4 cider does not exist, then we should let the bottlerocket host determine the cluster-dns-ip itself by not setting `cluster-dns-ip` in the instance userdata.